test/converters.test.ts

```diff
@@ -124,7 +126,7 @@ test("decimal converter", async () => {
   await checkWithOptions(converters.decimal({}), "43,14", "43.14", {
     decimalSeparator: ","
   });
-  await checkWithOptions(converters.decimal({}), "4.314.314", "4314314", {
+  await failsWithOptions(converters.decimal({}), "4.314.314", {
```

Comment:
This is a test for the wrong thing -- you don't want to test that the converter always fails in this case, instead you want to stop this situation from being allowed in the first place. The problem is that it's too late once the converter is actually in use. Instead we should throw an error immediately when the user tries to pass this these options into the state. In addition (as an extra safeguard, but less useful as later during runtime) you could add in a throw when this situation is detected in the converter rawValidate (not a `return false`; this is not a recoverable error).